### PR TITLE
[DOC] Tweaks for Array#shuffle

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.rb [hunk 1]
```diff
@@ -45,15 +45,22 @@ def each
   end
 
   # call-seq:
-  #    array.shuffle!(random: Random) -> array
+  #   shuffle(random: Random) -> new_array
```
Comment (peterzhu2118) [inline array.rb:48]: This is documentation for `Array#shuffle!`, not `Array#shuffle`.

```suggestion
  #   shuffle!(random: Random) -> new_array
```

File: array.rb [hunk 2]
```diff
@@ -45,15 +45,22 @@ def each
...
   #
-  # The optional +random+ argument will be used as the random number generator:
-  #    a.shuffle!(random: Random.new(1))  #=> [1, 3, 2]
+  #   a =            [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  #   a.shuffle # => [0, 8, 1, 9, 6, 3, 4, 7, 2, 5]
+  #   a.shuffle # => [8, 9, 0, 5, 1, 2, 6, 4, 7, 3]
+  #
+  #   a =            [0, 1, 0, 1, 0, 1, 0, 1, 0, 1]
+  #   a.shuffle # => [1, 0, 1, 1, 0, 0, 1, 0, 0, 1]
+  #   a.shuffle # => [1, 1, 0, 0, 0, 1, 1, 0, 0, 1]
```
Comment (peterzhu2118) [inline array.rb:59]: These examples are all for `Array#shuffle` and not `Array#shuffle!`.

File: array.rb [hunk 3]
```diff
@@ -45,15 +45,22 @@ def each
...
 
   # call-seq:
-  #    array.shuffle!(random: Random) -> array
+  #   shuffle(random: Random) -> new_array
   #
-  # Shuffles the elements of +self+ in place.
-  #    a = [1, 2, 3] #=> [1, 2, 3]
-  #    a.shuffle!    #=> [2, 3, 1]
-  #    a             #=> [2, 3, 1]
+  # Returns a new array containing all elements from +self+ in a random order,
```
Comment (peterzhu2118) [inline array.rb:50]: It does not return a new array, it modifies in place.

**Comment by BurdetteLamar** [comment]
I meant for this documentation to be for `Array#shuffle`, not `Array#shuffle!`;  I put it in the wrong place.  Will try to fix it.

**Comment by BurdetteLamar** [comment]
Will re-submit.

**Pull request closed without merging**